staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -524,14 +526,16 @@ func (a *Authenticator) wrapCmdRunErrorLocked(err error) error {
 
 		return errors.New(builder.String())
 
-	case *exec.ExitError: // Binary execution failed (see exec.Cmd.Run()).
-		e := err.(*exec.ExitError)
+	case errors.As(err, &exitError): // Binary execution failed (see exec.Cmd.Run()).
 		return fmt.Errorf(
-			"exec: executable %s failed with exit code %d",
+			"exec: binary %s failed with exit code %d",
 			a.cmd,
-			e.ProcessState.ExitCode(),
+			exitError.ProcessState.ExitCode(),
 		)
 
+	case os.IsPermission(err):
+		return fmt.Errorf("exec: binary %s not executable", a.cmd)
```

Comment:
I'm not in love with changing this error message to say "binary" instead of "executable" (because the plugin could be a bash script, which I would not consider to be a binary), but I wanted to avoid the redundancy of the error message "executable blah is not executable". Any suggestions?

Reply:
A new unit test message `binary ./testdata/not-executable.sh not executable` seems a little odd as @ankeesler pointed at the above.
How about keeping "executable" for the other places and removing "executable" from the above case `os.IsPermission` only to avoid unnecessary confusions?